src/inputs/Autocomplete.tsx

```diff
@@ -0,0 +1,136 @@
+import { Key, ReactNode, useRef } from "react";
+import { useComboBox, useOverlayPosition } from "react-aria";
+import { Item, useComboBoxState } from "react-stately";
+import { Icon } from "src/components";
+import { Popover } from "src/components/internal";
+import { PresentationFieldProps } from "src/components/PresentationContext";
+import { ListBox } from "src/inputs/internal/ListBox";
+import { TextFieldBase, TextFieldBaseProps } from "src/inputs/TextFieldBase";
+import { valueToKey } from "src/inputs/Value";
+
+interface AutocompleteProps<T>
+  extends Pick<PresentationFieldProps, "labelStyle">,
+    Pick<TextFieldBaseProps<any>, "label" | "clearable" | "startAdornment"> {
+  onSelect: (item: T) => void;
+  /** A function that returns how to render the an option in the menu. If not set, `getOptionLabel` will be used */
+  getOptionMenuLabel?: (o: T) => ReactNode;
+  /** A function that returns the string value of the option. Used for accessibility purposes */
+  getOptionLabel: (o: T) => string;
+  /** A function that returns a unique key for an option */
+  getOptionValue: (o: T) => string;
+  /** Called when the input value changes */
+  onChange: (value: string) => void;
+  /** The current value of the input */
+  value: string | undefined;
```

Comment:
Thought: Wdyt about keeping this `value` internally controlled, like our own internal `useState`, and then firing the `onChange` but debounced?

That way the caller doesn't have to do their own debounce / keep track of the every-single-key-press value of our text field.

Reply:
I thought about this. I still have another example of this component that uses the `useAsyncList` from React-Aria, which makes this `value` internally controlled. Then it handles its own debouncing, fetching, and list of results. I was thinking of making the component smarter later... like you can either have it completely controlled (as-is right now), or more uncontrolled where you just pass in the fetch function and the rest is handled. 
I can see the benefit of adding the debounce here, but then I wonder if that'll hinder the Algolia search experience. I would assume they must do some sort of debouncing on their end.

Reply:
If you feel really strongly about doing the debouncing locally, we can. I just don't see it as a pressing need at the moment. Personally I think having the debouncing happen closer to the fetch call would be ideal. It'd just be more clear when/where the debounce is happening. Otherwise, I think I'd want the prop to maybe be named `onDebouncedInputChange` and allow the user to specify a `props.debounceMs` or something. 
Also makes me wonder if there'd be a time where we wouldn't want to debounce? I guess if the potential options are all local (doesn't require an api fetch), then the debounce wouldn't be necessary. 

Anyways, let me know what you think. I'm leaning towards keeping it as is.

Reply:
Cool, I'm good with this. We can always change it if it becomes obvious we have a ton of callers doing the same debounce.